### PR TITLE
usb: device: fix unaligned memory access in Audio class

### LINKED ISSUE #78920: USB Audio: Unaligned address data access
https://github.com/zephyrproject-rtos/zephyr/blob/810f978bed4fbc35eec165eaaedd17745b1ec255/subsys/usb/device/class/audio/audio.c#L598
https://github.com/zephyrproject-rtos/zephyr/blob/810f978bed4fbc35eec165eaaedd17745b1ec255/subsys/usb/device/class/audio/audio.c#L580
https://github.com/zephyrproject-rtos/zephyr/blob/810f978bed4fbc35eec165eaaedd17745b1ec255/subsys/usb/device/class/audio/audio.c#L592
  
https://github.com/zephyrproject-rtos/zephyr/blob/810f978bed4fbc35eec165eaaedd17745b1ec255/samples/subsys/usb/audio/headphones_microphone/src/main.c#L56
https://github.com/zephyrproject-rtos/zephyr/blob/810f978bed4fbc35eec165eaaedd17745b1ec255/samples/subsys/usb/audio/headset/src/main.c#L54

`int16_t` access to unaligned memory, which will cause exceptions on some platforms, such as mips

### PR DESCRIPTION
**Pull request by LuoZhongYao**
Use `UNALIGNED_*`, `sys_get_*` to fix unaligned memory access issues
fix #78920

### CODE REVIEW

File: samples/subsys/usb/audio/headphones_microphone/src/main.c [hunk 1]
```diff
@@ -53,7 +53,7 @@ static void feature_update(const struct device *dev,
 	case USB_AUDIO_FU_MUTE_CONTROL:
 		break;
 	case USB_AUDIO_FU_VOLUME_CONTROL:
-		volume = *((int16_t *)(evt->val));
+		volume = UNALIGNED_GET((int16_t *)evt->val);
```
Comment (jfischer-no) [inline samples/subsys/usb/audio/headphones_microphone/src/main.c:56]: Would it work with clang and why not just `memcpy(&volume, evt->val, sizeof(volume));`?

Comment (LuoZhongYao) [reply samples/subsys/usb/audio/headphones_microphone/src/main.c:56]: My development environment is not convenient for testing clang (soc is a customized mips),  but zephyr also uses UNALIGNED_GET in many places, and `include/zephyr/toolchain/llvm.h` includes `include/zephyr/toolchain/gcc.h`, so clang should also be able to use it. If you have any doubts about this, I can resubmit using memcpy

**Review by tmon-nordic — APPROVED** [review]

File: samples/subsys/usb/audio/headphones_microphone/src/main.c (hunk 1 above)
Comment (jfischer-no) [reply samples/subsys/usb/audio/headphones_microphone/src/main.c:56]: Looks like the __typeof__ should work there too https://clang.llvm.org/docs/UsersManual.html

**Review by jfischer-no — COMMENTED** [review]
@LuoZhongYao Please fix compliance issues, you commit title should be like: `usb: device: fix unaligned memory access in Audio class`

**Comment by kartben** [comment]
> @LuoZhongYao Please fix compliance issues, you commit title should be like: `usb: device: fix unaligned memory access in Audio class`

@jfischer-no this has been addressed - please revisit

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**